internal/compiler/langtype.rs

```diff
@@ -917,7 +912,7 @@ impl StructName {
     pub fn or(self, other: Self) -> Self {
         match self {
             Self::None => other,
-            this @ _ => this,
+            this => this,
```

Comment:
This is one of the thing i disagree with clippy on.

To me, the `foo @ _ =>` makes the distinction between creating a new variable foo, or matching to an existing (const) variable foo. It's true that const would be written upper case, but still, i feel the `@` is more readable. Anyway, if clippy insists on it, what can i say? ;-)